runtime/protocol/http-client-engines/http-client-engine-crt/common/src/aws/smithy/kotlin/runtime/http/engine/crt/RequestUtil.kt

```diff
@@ -60,11 +60,7 @@ internal fun HttpRequest.toCrtRequest(callContext: CoroutineContext): aws.sdk.ko
         headers.forEach { key, values -> appendAll(key, values) }
     }
 
-    val bodyLen = body.contentLength
-    val contentLength = when {
-        bodyLen != null -> if (bodyLen > 0) bodyLen.toString() else null
-        else -> headers[CONTENT_LENGTH_HEADER]
-    }
+    val contentLength = body.contentLength?.toString() ?: headers[CONTENT_LENGTH_HEADER]
```

Comment:
**Question**: This is interesting...for CRT we fallback to an explicitly-provided `Content-length` header if one was provided.

It looks like OkHttp only derives content length from `RequestBody.contentLength`. If `RequestBody.contentLength == -1` (unknown), it [explicitly removes any previously-set `Content-length` header](https://github.com/square/okhttp/blob/ee1ba4656d80ed62485387217271c7fa31dc888e/okhttp/src/jvmMain/kotlin/okhttp3/internal/http/BridgeInterceptor.kt#L52-L55). Should we be falling back to the user-set `Content-length` header in OkHttp for cases where body length is unspecified?

Reply:
I think we should. I checked if sending a `Content-Length` would conflict with chunked requests, but it's purely additive and *shouldn't* clash with chunked requests according to our internal spec. 

I can't imagine a case where the fallback behavior would be harmful

Reply:
What if they set content length in an interceptor before becoming a chunked request?

Reply:
I think that's still fine. A chunked request can have a content length header